### ISSUE TITLE
TWE Design System Components: Update Buttons according to M3

### ISSUE THREAD

**Issue by mklmrgn** [hackforla/internship#619]
### Overview
Change the Button Components to a more modern style based on Material Design 3.

### Action Items
- [x] Research button styling for Google's M3
- [x] Update Component Design
- [x] Delete extraneous variant states (ie enabled vs active)
- [x] Create buttons with icons
- [x] Communicate changes in issue's comments and with design team
- [ ] Get sign-off from Project Manager
- [x] Update Component Library and Microsite
- [ ] Close Issue
- [ ] Release Dependency on https://github.com/hackforla/internship/issues/409 and if it is the last dependency to get released, move issue to the new issue approval column & add the `ready for design lead` label

### Resources/Instructions
Use Material Design (https://m3.material.io/) as guideline

**Comment by KC-skc** [comment]
@muntiCloud Please provide update

1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by JaimeCucalon** [comment]
This week I researched the M3 styling and approach. I learned that they have quite an extensive level of buttons and CTA elements that determine hierarchy for call to action. I'm wondering if we can narrow down the level of buttons before getting started. And focused on the common buttons. 
<img width="984" alt="Screenshot 2024-09-26 at 12 49 00 PM" src="https://github.com/user-attachments/assets/47ecdd7c-96a8-471a-8624-dccb0aae78d6">
vs
<img width="1024" alt="lvlz8hsu-1" src="https://github.com/user-attachments/assets/86a296af-4457-491a-a3d7-e5845455af92">

I also studied the existing button master for the internship team. I believe it's minimal but not flexible enough. I would love to the discuss the complexity of buttons we want to keep for the project without adding too much work for future updates.

**Comment by JaimeCucalon** [comment]
https://www.figma.com/design/TTRS2FWXsrymHYpPJL1IdH/Internship-Team-Main-file?node-id=6474-7596&t=dsy9ee4753S2UBxn-4

1. Progress:
- Created 4 style of buttons following MD3 Specs and guidelines-Filled, outlined, text, and tonal. 
- Created icon variants for all 4 with boolean properties

2. Blockers: 
- Researching if the pressed state is needed for our design system. It is very similar to the focused state, main difference is that it uses a **ripple animation**. Need to check with dev team to confirm feasibility.
- Waiting on updated styles for icons, text, and button
- Need to add icon variants to the design system. Color variants, and size variants

3. Availability - 4 hours

4. ETA: Hoping to have this component ready by end of next week.

**Comment by JaimeCucalon** [comment]
Progress:
Matched elevation, text size, icon size, and padding to MD3 design. There is one small icon/padding detail missing that needs addressing.

Blockers:
Researching the use of different size buttons by MD3. So Only the FAB buttons have three different sizes - small, regular, and large - https://m3.material.io/components/floating-action-button/overview

Availability - 4 hours

ETA: Need another week.

**Comment by JaimeCucalon** [comment]
Progress:
- Revised all the color applications based on new color styles and state layers from Michael
- Set up the icon property to be swappable without changing color. Relabelled all icons in the library to fix this minor glitch

![Uploading Screenshot 2024-10-24 at 9.41.39 AM.png…]()

Blockers:
- Need to confirm the color application is correct before creating different size buttons

Availability - 4 hours

ETA: Next Tuesday

**Comment by JaimeCucalon** [comment]
Made some additional changes to the button component colors in order to pass the color contrast requirements. 
[The page includes two versions](https://www.figma.com/design/TTRS2FWXsrymHYpPJL1IdH/Internship-Team-Main-file?node-id=2-72)
- Version **on the left** follows the MD3 styling 100% but fails the color contrast. Each one of the button states shows the color contrast ratings
- Version **on the right** has two slight changes to the MD3 styling but it passes color contrast. More details in the figma file for changes made. 
- The only button states that don't pass the color contrast standards is the disabled state. This is also true for the MD3 design system button component

@mklmrgn - Ready for review
<img width="652" alt="Screenshot 2024-10-24 at 12 04 41 PM" src="https://github.com/user-attachments/assets/d4ca67fc-212d-4362-a3f4-dad9b406f0be">

**Comment by JaimeCucalon** [comment]
**Progress:**
- Worked on the color contrast of all buttons and states. Discussed with Michael changes required to the color system in order to pass color contrast requirements.
<img width="1200" alt="Screenshot 2024-10-31 at 9 48 23 AM" src="https://github.com/user-attachments/assets/180774e7-4ffd-4ac4-950e-69ad5335ef5e">

**Blockers:**
Need to determine if our color system requires different size buttons to create hierarchy for calls to action on the site. The new button system has 4 type of buttons; Filled, Outline, Tonal, and Text. This follows the MD3 system. 
<img width="551" alt="Screenshot 2024-10-31 at 9 47 10 AM" src="https://github.com/user-attachments/assets/956d9202-c013-4915-b9a7-400696eaf563">

vs the current design system on the site that has 3 sizes, and doesn't follow the MD3 system closely. 
<img width="755" alt="Screenshot 2024-10-31 at 9 40 43 AM" src="https://github.com/user-attachments/assets/f9f993ac-9cce-48fe-b77c-518a4045921f">

**Comment by JaimeCucalon** [comment]
**Notes from TWE on Button sizes:**
The discussion with Bonnie and Michael revealed that we have different button sizes because it follows the HFLA design system. That system showed that there are some large spaces where the primary call to action needs to be larger in order to fill the space for hero banners. 

**We decided to look into different sizes for the primary button only**. Since that will be the button that gets featured the most. The secondary and tertiary button types will only have one size. 

Here are the examples we reviewed that showed how both HFLA and MD3 use a similar approach. 
<img width="827" alt="Screenshot 2024-10-31 at 10 32 54 AM" src="https://github.com/user-attachments/assets/7a79d886-73d5-488e-942f-a2f9010ad23c">
<img width="888" alt="Screenshot 2024-10-31 at 10 24 29 AM" src="https://github.com/user-attachments/assets/670a7d6c-c2db-442f-a781-cfdf5bdd5d3e">

**Comment by ExperimentsInHonesty** [comment]
@JaimeCucalon  Please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by JaimeCucalon** [comment]
**Progress:**
Reviewed the Hack for LA design system and website. From those learnings, I added an **extra large button size** to the Primary Button Component. This should suffice large hero banner spaces. 

Also, broke up the button into two separate components, Primary Button and Secondary Button with respective variants.
<img width="644" alt="Screenshot 2024-11-14 at 9 30 46 AM" src="https://github.com/user-attachments/assets/1865d0b3-32c6-4164-936b-94e7220467c7">

**Blockers**
Waiting on approval to make these the final edits to the button sizes and variants. Also, need to update the color stylings based on recent changes from Mike. 

**Availability**
4 hours

**ETA**
Expected to complete it by end of November latest. 

https://www.figma.com/design/TTRS2FWXsrymHYpPJL1IdH/Internship-Team-Main-file?node-id=6474-7596&t=dsy9ee4753S2UBxn-4

**Comment by ExperimentsInHonesty** [comment]
@mklmrgn This issue is ready for your review.  Jaime mentioned the following in the comment above

>Waiting on approval to make these the final edits to the button sizes and variants. Also, need to update the color stylings based on recent changes from Mike.

**Comment by JaimeCucalon** [comment]
**Progress:**
Received approval for the sizes and color styling for the button variants. And combined the primary and secondary button variants into one component. 

Next step is to create guidelines, and web page specs frame. 

**Blockers:**
None

**ETA:**
Expected to complete by next week. 
<img width="639" alt="Screenshot 2024-11-21 at 9 43 43 AM" src="https://github.com/user-attachments/assets/c6267cad-f701-4538-8c49-b21721743c2a">

**Comment by mklmrgn** [comment]
Progress:
Updated specs and replaced old components on Design System (WIP)[ Figma Page](https://www.figma.com/design/TTRS2FWXsrymHYpPJL1IdH/Internship-Team-Main-file?node-id=5505-13300&t=ox3xQJfHo99vIEbS-1)

![Image](https://github.com/user-attachments/assets/626fb797-5782-4841-b8fc-413f4c52f48f)

Next step is to update microsite with changes

Blockers:
None

ETA:
Expected to complete by next week.

**Comment by mklmrgn** [comment]
Progress:
Updated microsite with updated components - Component and microsite are complete

![Image](https://github.com/user-attachments/assets/9c2827c1-d037-4222-a724-89ae2f18d305)

Blockers:
None

**Comment by mklmrgn** [comment]
@joshfishman This is ready for Dev Lead sign off

**Comment by mklmrgn** [comment]
Added Dark Mode buttons

![Image](https://github.com/user-attachments/assets/add938f4-bf02-4b3f-8ab5-f0f5684c789e)
